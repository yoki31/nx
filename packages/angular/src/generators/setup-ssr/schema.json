{
  "$schema": "https://json-schema.org/schema",
  "$id": "GeneratorAngularUniversalSetup",
  "cli": "nx",
  "title": "Generate Angular Universal (SSR) setup for an Angular App",
  "description": "Create the additional configuration required to enable SSR via Angular Universal for an Angular application.",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the application to generate the Angular Universal configuration for.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What app would you like to generate an Angular Universal configuration for?",
      "x-dropdown": "projects"
    },
    "appId": {
      "type": "string",
      "format": "html-selector",
      "description": "The `appId` to use with `withServerTransition`.",
      "default": "serverApp",
      "x-deprecated": "This is deprecated and ignored since Angular 16 and not supported since Angular 17."
    },
    "main": {
      "type": "string",
      "format": "path",
      "description": "The name of the main entry-point file.",
      "default": "main.server.ts"
    },
    "serverFileName": {
      "type": "string",
      "default": "server.ts",
      "description": "The name of the Express server file."
    },
    "serverPort": {
      "type": "number",
      "default": 4000,
      "description": "The port for the Express server."
    },
    "rootModuleFileName": {
      "type": "string",
      "format": "path",
      "description": "The name of the root module file",
      "default": "app.server.module.ts"
    },
    "rootModuleClassName": {
      "type": "string",
      "description": "The name of the root module class.",
      "default": "AppServerModule"
    },
    "standalone": {
      "type": "boolean",
      "description": "Use Standalone Components to bootstrap SSR."
    },
    "hydration": {
      "type": "boolean",
      "description": "Set up Hydration for the SSR application. It defaults to `true` for Angular versions >= 17.0.0. Otherwise, it defaults to `false`."
    },
    "skipFormat": {
      "type": "boolean",
      "description": "Skip formatting the workspace after the generator completes.",
      "x-priority": "internal"
    },
    "skipPackageJson": {
      "type": "boolean",
      "default": false,
      "description": "Do not add dependencies to `package.json`.",
      "x-priority": "internal"
    }
  },
  "required": ["project"],
  "additionalProperties": false
}
