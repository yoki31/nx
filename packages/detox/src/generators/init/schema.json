{
  "$schema": "https://json-schema.org/schema",
  "title": "Add Detox Schematics",
  "description": "Add Detox Schematics.",
  "type": "object",
  "properties": {
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false,
      "x-priority": "internal"
    },
    "skipPackageJson": {
      "type": "boolean",
      "default": false,
      "description": "Do not add dependencies to `package.json`.",
      "x-priority": "internal"
    },
    "keepExistingVersions": {
      "type": "boolean",
      "x-priority": "internal",
      "description": "Keep existing dependencies versions",
      "default": false
    },
    "updatePackageScripts": {
      "type": "boolean",
      "x-priority": "internal",
      "description": "Update `package.json` scripts with inferred targets",
      "default": false
    }
  },
  "required": []
}
