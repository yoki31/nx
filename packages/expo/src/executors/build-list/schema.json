{
  "version": 2,
  "outputCapture": "direct-nodejs",
  "$schema": "https://json-schema.org/schema",
  "$id": "NxExpoEasBuildList",
  "cli": "nx",
  "title": "Expo EAS Build List executor",
  "description": "List all Expo Application Services (EAS) builds for your Expo project.",
  "type": "object",
  "properties": {
    "platform": {
      "enum": ["ios", "android", "all"],
      "alias": "p",
      "description": "The platform to build the app.",
      "x-priority": "important"
    },
    "json": {
      "type": "boolean",
      "description": "Enable JSON output, non-JSON messages will be printed to stderr"
    },
    "interactive": {
      "type": "boolean",
      "description": "Run the command in interactive mode."
    },
    "status": {
      "enum": [
        "new",
        "in-queue",
        "in-progress",
        "errored",
        "finished",
        "canceled"
      ],
      "description": "Status of EAS build"
    },
    "distribution": {
      "enum": ["store", "internal", "simulator"],
      "description": "Distribution of EAS build"
    },
    "channel": {
      "type": "string",
      "description": "Channel of EAS build"
    },
    "appVersion": {
      "type": "string",
      "description": "App version of EAS build"
    },
    "appBuildVersion": {
      "type": "string",
      "description": "App build version of EAS build"
    },
    "sdkVersion": {
      "type": "string",
      "description": "SDK version of EAS build"
    },
    "runtimeVersion": {
      "type": "string",
      "description": "Runtime version of EAS build"
    },
    "appIdentifier": {
      "type": "string",
      "description": "App identifier of EAS build"
    },
    "buildProfile": {
      "type": "string",
      "description": "Build profile of EAS build"
    },
    "gitCommitHash": {
      "type": "string",
      "description": "Git commit hash of EAS build"
    },
    "limit": {
      "type": "number",
      "description": "Limit of numbers to list EAS builds"
    }
  },
  "required": [],
  "examplesFile": "../../../docs/build-list-examples.md"
}
