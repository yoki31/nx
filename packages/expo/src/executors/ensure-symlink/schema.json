{
  "version": 2,
  "outputCapture": "direct-nodejs",
  "cli": "nx",
  "$id": "NxExpoEnsureSymlink",
  "$schema": "https://json-schema.org/schema",
  "title": "Ensure Symlink for Expo",
  "description": "Ensure workspace node_modules is symlink under app's node_modules folder.",
  "type": "object",
  "properties": {}
}
