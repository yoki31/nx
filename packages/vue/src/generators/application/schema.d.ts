import type { ProjectNameAndRootFormat } from '@nx/devkit/src/generators/project-name-and-root-utils';
import type { Linter } from '@nx/eslint';

export interface Schema {
  name: string;
  style: 'none' | 'css' | 'scss' | 'less';
  skipFormat?: boolean;
  directory?: string;
  projectNameAndRootFormat?: ProjectNameAndRootFormat;
  tags?: string;
  unitTestRunner?: 'vitest' | 'none';
  inSourceTests?: boolean;
  e2eTestRunner: 'cypress' | 'playwright' | 'none';
  linter: Linter;
  routing?: boolean;
  js?: boolean;
  strict?: boolean;
  setParserOptionsProject?: boolean;
  skipPackageJson?: boolean;
  rootProject?: boolean;
  addPlugin?: boolean;
}

export interface NormalizedSchema extends Schema {
  projectName: string;
  appProjectRoot: string;
  e2eProjectName: string;
  e2eProjectRoot: string;
  parsedTags: string[];
  devServerPort?: number;
}
