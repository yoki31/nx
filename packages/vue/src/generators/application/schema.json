{
  "$schema": "https://json-schema.org/schema",
  "cli": "nx",
  "$id": "NxVueApp",
  "title": "Create a Vue Application",
  "description": "Create a Vue application for Nx.",
  "examples": [
    {
      "command": "nx g app myapp --directory=myorg/myapp",
      "description": "Generate `apps/myorg/myapp` and `apps/myorg/myapp-e2e`"
    },
    {
      "command": "nx g app myapp --routing",
      "description": "Set up Vue Router"
    }
  ],
  "type": "object",
  "properties": {
    "name": {
      "description": "The name of the application.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the application?",
      "pattern": "^[a-zA-Z][^:]*$"
    },
    "directory": {
      "description": "The directory of the new application.",
      "type": "string",
      "alias": "dir",
      "x-priority": "important"
    },
    "projectNameAndRootFormat": {
      "description": "Whether to generate the project name and root directory as provided (`as-provided`) or generate them composing their values and taking the configured layout into account (`derived`).",
      "type": "string",
      "enum": ["as-provided", "derived"]
    },
    "style": {
      "description": "The file extension to be used for style files.",
      "type": "string",
      "default": "css",
      "alias": "s",
      "x-prompt": {
        "message": "Which stylesheet format would you like to use?",
        "type": "list",
        "items": [
          {
            "value": "css",
            "label": "CSS"
          },
          {
            "value": "scss",
            "label": "SASS(.scss)       [ https://sass-lang.com                     ]"
          },
          {
            "value": "less",
            "label": "LESS              [ https://lesscss.org                       ]"
          },
          {
            "value": "none",
            "label": "None"
          }
        ]
      }
    },
    "linter": {
      "description": "The tool to use for running lint checks.",
      "type": "string",
      "enum": ["eslint", "none"],
      "default": "eslint"
    },
    "routing": {
      "type": "boolean",
      "description": "Generate application with routes.",
      "x-prompt": "Would you like to add Vue Router to this application?",
      "default": false
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false,
      "x-priority": "internal"
    },
    "unitTestRunner": {
      "type": "string",
      "enum": ["vitest", "none"],
      "description": "Test runner to use for unit tests.",
      "x-prompt": "Which unit test runner would you like to use?",
      "default": "vitest"
    },
    "inSourceTests": {
      "type": "boolean",
      "default": false,
      "description": "When using Vitest, separate spec files will not be generated and instead will be included within the source files. Read more on the Vitest docs site: https://vitest.dev/guide/in-source.html"
    },
    "e2eTestRunner": {
      "type": "string",
      "enum": ["playwright", "cypress", "none"],
      "description": "Test runner to use for end to end (E2E) tests.",
      "x-prompt": "Which E2E test runner would you like to use?",
      "default": "playwright"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the application (used for linting).",
      "alias": "t"
    },
    "js": {
      "type": "boolean",
      "description": "Generate JavaScript files rather than TypeScript files.",
      "default": false
    },
    "strict": {
      "type": "boolean",
      "description": "Whether to enable tsconfig strict mode or not.",
      "default": true
    },
    "setParserOptionsProject": {
      "type": "boolean",
      "description": "Whether or not to configure the ESLint `parserOptions.project` option. We do not do this by default for lint performance reasons.",
      "default": false
    },
    "skipPackageJson": {
      "description": "Do not add dependencies to `package.json`.",
      "type": "boolean",
      "default": false,
      "x-priority": "internal"
    },
    "rootProject": {
      "description": "Create a application at the root of the workspace",
      "type": "boolean",
      "default": false,
      "hidden": true
    }
  },
  "required": ["name"],
  "examplesFile": "../../../docs/application-examples.md"
}
