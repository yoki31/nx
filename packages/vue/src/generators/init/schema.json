{
  "$schema": "https://json-schema.org/schema",
  "$id": "NxVueInit",
  "title": "Init Vue Plugin",
  "description": "Initialize a Vue Plugin.",
  "cli": "nx",
  "type": "object",
  "properties": {
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    },
    "skipPackageJson": {
      "description": "Do not add dependencies to `package.json`.",
      "type": "boolean",
      "default": false
    },
    "keepExistingVersions": {
      "type": "boolean",
      "x-priority": "internal",
      "description": "Keep existing dependencies versions",
      "default": false
    }
  },
  "required": []
}
