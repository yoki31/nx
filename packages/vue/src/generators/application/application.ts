import {
  addProjectConfiguration,
  formatFiles,
  GeneratorCallback,
  runTasksInSerial,
  toJS,
  Tree,
} from '@nx/devkit';
import { Linter } from '@nx/eslint';
import { initGenerator as jsInitGenerator } from '@nx/js';
import { Schema } from './schema';
import { normalizeOptions } from './lib/normalize-options';
import { vueInitGenerator } from '../init/init';
import { addLinting } from '../../utils/add-linting';
import { addE2e } from './lib/add-e2e';
import { createApplicationFiles } from './lib/create-application-files';
import { addVite } from './lib/add-vite';
import { extractTsConfigBase } from '../../utils/create-ts-config';
import { ensureDependencies } from '../../utils/ensure-dependencies';
import { logShowProjectCommand } from '@nx/devkit/src/utils/log-show-project-command';

export function applicationGenerator(tree: Tree, options: Schema) {
  return applicationGeneratorInternal(tree, { addPlugin: false, ...options });
}

export async function applicationGeneratorInternal(
  tree: Tree,
  _options: Schema
): Promise<GeneratorCallback> {
  const options = await normalizeOptions(tree, _options);
  options.addPlugin ??= process.env.NX_ADD_PLUGINS !== 'false';

  const tasks: GeneratorCallback[] = [];

  addProjectConfiguration(tree, options.name, {
    root: options.appProjectRoot,
    projectType: 'application',
    sourceRoot: `${options.appProjectRoot}/src`,
    targets: {},
  });

  tasks.push(
    await jsInitGenerator(tree, {
      ...options,
      tsConfigName: options.rootProject
        ? 'tsconfig.json'
        : 'tsconfig.base.json',
      skipFormat: true,
    })
  );
  tasks.push(
    await vueInitGenerator(tree, {
      ...options,
      skipFormat: true,
    })
  );
  if (!options.skipPackageJson) {
    tasks.push(ensureDependencies(tree, options));
  }

  if (!options.rootProject) {
    extractTsConfigBase(tree);
  }

  createApplicationFiles(tree, options);

  tasks.push(
    await addLinting(
      tree,
      {
        name: options.projectName,
        projectRoot: options.appProjectRoot,
        linter: options.linter ?? Linter.EsLint,
        unitTestRunner: options.unitTestRunner,
        skipPackageJson: options.skipPackageJson,
        setParserOptionsProject: options.setParserOptionsProject,
        rootProject: options.rootProject,
        addPlugin: options.addPlugin,
      },
      'app'
    )
  );

  tasks.push(await addVite(tree, options));

  tasks.push(await addE2e(tree, options));

  if (options.js) toJS(tree);

  if (!options.skipFormat) await formatFiles(tree);

  tasks.push(() => {
    logShowProjectCommand(options.projectName);
  });

  return runTasksInSerial(...tasks);
}

export default applicationGenerator;
