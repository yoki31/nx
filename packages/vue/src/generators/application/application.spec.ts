import 'nx/src/internal-testing-utils/mock-project-graph';

import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { Tree, readProjectConfiguration } from '@nx/devkit';

import { applicationGenerator } from './application';
import { Schema } from './schema';

describe('application generator', () => {
  let tree: Tree;
  const options: Schema = { name: 'test' } as Schema;

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();
  });

  it('should run successfully', async () => {
    await applicationGenerator(tree, options);
    const config = readProjectConfiguration(tree, 'test');
    expect(config).toBeDefined();
  });

  it('should set up project correctly with given options', async () => {
    await applicationGenerator(tree, { ...options, unitTestRunner: 'vitest' });
    expect(tree.read('.eslintrc.json', 'utf-8')).toMatchSnapshot();
    expect(tree.read('test/vite.config.ts', 'utf-8')).toMatchSnapshot();
    expect(tree.read('test/.eslintrc.json', 'utf-8')).toMatchSnapshot();
    expect(tree.read('test/src/app/App.spec.ts', 'utf-8')).toMatchSnapshot();
    expect(listFiles(tree)).toMatchSnapshot();
  });

  it('should not use stylesheet if --style=none', async () => {
    await applicationGenerator(tree, { ...options, style: 'none' });

    expect(tree.exists('test/src/style.none')).toBeFalsy();
    expect(tree.read('test/src/main.ts', 'utf-8')).not.toContain('styles.none');
  });
});

function listFiles(tree: Tree): string[] {
  const files = new Set<string>();
  tree.listChanges().forEach((change) => {
    if (change.type !== 'DELETE') {
      files.add(change.path);
    }
  });

  return Array.from(files).sort((a, b) => a.localeCompare(b));
}
