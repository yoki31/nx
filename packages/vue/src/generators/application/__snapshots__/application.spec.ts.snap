// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application generator should set up project correctly with given options 1`] = `
"{
  "root": true,
  "ignorePatterns": ["**/*"],
  "plugins": ["@nx"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx", "*.vue"],
      "rules": {
        "@nx/enforce-module-boundaries": [
          "error",
          {
            "enforceBuildableLibDependency": true,
            "allow": [],
            "depConstraints": [
              {
                "sourceTag": "*",
                "onlyDependOnLibsWithTags": ["*"]
              }
            ]
          }
        ]
      }
    },
    {
      "files": ["*.ts", "*.tsx"],
      "extends": ["plugin:@nx/typescript"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "extends": ["plugin:@nx/javascript"],
      "rules": {}
    }
  ]
}
"
`;

exports[`application generator should set up project correctly with given options 2`] = `
"/// <reference types='vitest' />
import { defineConfig } from 'vite';
import vue from '@vitejs/plugin-vue';
import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';

export default defineConfig({
  root: __dirname,
  cacheDir: '../node_modules/.vite/test',

  server: {
    port: 4200,
    host: 'localhost',
  },

  preview: {
    port: 4300,
    host: 'localhost',
  },

  plugins: [vue(), nxViteTsPaths()],

  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [ nxViteTsPaths() ],
  // },

  build: {
    outDir: '../dist/test',
    emptyOutDir: true,
    reportCompressedSize: true,
    commonjsOptions: {
      transformMixedEsModules: true,
    },
  },

  test: {
    watch: false,
    globals: true,
    cache: {
      dir: '../node_modules/.vitest/test',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],

    reporters: ['default'],
    coverage: {
      reportsDirectory: '../coverage/test',
      provider: 'v8',
    },
  },
});
"
`;

exports[`application generator should set up project correctly with given options 3`] = `
"{
  "extends": [
    "plugin:vue/vue3-essential",
    "eslint:recommended",
    "@vue/eslint-config-typescript",
    "@vue/eslint-config-prettier/skip-formatting",
    "../.eslintrc.json"
  ],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx", "*.vue"],
      "rules": {
        "vue/multi-word-component-names": "off"
      }
    }
  ]
}
"
`;

exports[`application generator should set up project correctly with given options 4`] = `
"import { describe, it, expect } from 'vitest';

import { mount } from '@vue/test-utils';
import App from './App.vue';

describe('App', () => {
  it('renders properly', async () => {
    const wrapper = mount(App, {});

    expect(wrapper.text()).toContain('Welcome test ðŸ‘‹');
  });
});
"
`;

exports[`application generator should set up project correctly with given options 5`] = `
[
  ".eslintignore",
  ".eslintrc.json",
  ".prettierignore",
  ".prettierrc",
  ".vscode/extensions.json",
  "nx.json",
  "package.json",
  "test-e2e/.eslintrc.json",
  "test-e2e/playwright.config.ts",
  "test-e2e/project.json",
  "test-e2e/src/example.spec.ts",
  "test-e2e/tsconfig.json",
  "test/.eslintrc.json",
  "test/index.html",
  "test/project.json",
  "test/src/app/App.spec.ts",
  "test/src/app/App.vue",
  "test/src/app/NxWelcome.vue",
  "test/src/main.ts",
  "test/src/styles.css",
  "test/tsconfig.app.json",
  "test/tsconfig.json",
  "test/tsconfig.spec.json",
  "test/vite.config.ts",
  "tsconfig.base.json",
  "vitest.workspace.ts",
]
`;
